hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java

```diff
@@ -85,49 +90,18 @@ public void pureChecks(@NonNull final CryptoTransferTransactionBody op) throws P
-        for (final TokenTransferList tokenTransfer : tokenTransfers) {
-            final var tokenID = tokenTransfer.token();
-            tokenIds.add(tokenID);
-            validateTruePreCheck(tokenID != null && !tokenID.equals(TokenID.DEFAULT), INVALID_TOKEN_ID);
-
-            // Validate the fungible transfers
-            final var uniqueTokenAcctIds = new HashSet<Pair<AccountID, Boolean>>();
-            final var fungibleTransfers = tokenTransfer.transfers();
-            validateTruePreCheck(isNetZeroAdjustment(fungibleTransfers), TRANSFERS_NOT_ZERO_SUM_FOR_TOKEN);
-            boolean nonZeroFungibleValueFound = false;
-            for (final AccountAmount acctAmount : fungibleTransfers) {
-                validateTruePreCheck(acctAmount.hasAccountID(), INVALID_TRANSFER_ACCOUNT_ID);
-                uniqueTokenAcctIds.add(Pair.of(acctAmount.accountIDOrThrow(), acctAmount.isApproval()));
-                if (!nonZeroFungibleValueFound && acctAmount.amount() != 0) {
-                    nonZeroFungibleValueFound = true;
-                }
-            }
-            validateFalsePreCheck(
-                    uniqueTokenAcctIds.size() < fungibleTransfers.size(), ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS);
+        validateTokenTransfers(op.tokenTransfers());
+    }
 
-            // Validate the nft transfers
-            final var nftTransfers = tokenTransfer.nftTransfers();
-            nftIds.clear();
-            for (final NftTransfer nftTransfer : nftTransfers) {
-                validateTruePreCheck(nftTransfer.serialNumber() > 0, INVALID_TOKEN_NFT_SERIAL_NUMBER);
-                validateTruePreCheck(nftTransfer.hasSenderAccountID(), INVALID_TRANSFER_ACCOUNT_ID);
-                validateTruePreCheck(nftTransfer.hasReceiverAccountID(), INVALID_TRANSFER_ACCOUNT_ID);
-                validateFalsePreCheck(
-                        !nftIds.isEmpty() && nftIds.contains(nftTransfer.serialNumber()), INVALID_ACCOUNT_AMOUNTS);
-                validateFalsePreCheck(
-                        nftTransfer.senderAccountIDOrThrow().equals(nftTransfer.receiverAccountID()),
-                        ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS);
-                nftIds.add(nftTransfer.serialNumber());
-            }
-            // Verify that one and only one of the two types of transfers (fungible or non-fungible) is present
-            validateFalsePreCheck(
-                    uniqueTokenAcctIds.isEmpty() && nftIds.isEmpty(), EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS);
-        }
-        validateFalsePreCheck(tokenIds.size() < tokenTransfers.size(), TOKEN_ID_REPEATED_IN_TOKEN_LIST);
+    /**
+     * Performs pure checks that validates basic fields in the token airdrop transaction.
+     * @param op the token airdrop transaction body
+     * @throws PreCheckException if any of the checks fail
+     */
+    public void airdropsPureChecks(@NonNull final TokenAirdropTransactionBody op) throws PreCheckException {
```

Comment:
These should be in `airdropValidator`. Or have a base class for both. But having airdropPureChecks in `CryptoTransferValidator` name looks different

Reply:
This is fixed in the handler PR, as there was similar comment there.

Reply:
Ah got it. Thanks